SALTO-2468: bump Salesforce api to v55

tamtamirr opened the pull request:
Bumped Salesforce API from V50 to V55
---


---
_Release Notes_: 
Salesforce Adapter:
- Uses API V55

---
_User Notifications_: 
Big changes in Salesforce workspaces due to more data and correct types coming from the newer version of Salesforce

tamtamirr commented on packages/salesforce-adapter/src/filters/settings_type.ts:
```diff
@@ -86,6 +88,7 @@ const filterCreator: FilterCreator = ({ client, config }) => ({
         .map(info => getSettingsTypeName(info.fullName))
         .map(typeName => createSettingsType(client, typeName, knownTypes))
     )).flat()
+    _.pullAllBy(settingsTypes, objectTypes, e => e.elemID.getFullName())
```
The API changes caused certain types to appear twice due to this filter, this way I'm assuring they're unique.

ori-moisis replied:
did you check the information in those types? was it the same information?
do we still need this filter if we get the information through the "normal" fetch flow?
do we only get types? not the settings instances?

adamro replied:
I am missing something here. 
How did the API changes cause us to create two ObjectTypes? I don't fully understand this so I am not this solution is legit. Can you please elaborate?

tamtamirr replied:
@ori-moisis 
- Re information Yes, it's two certain type (`ObjectMapping` & `ObjectMappingField`), checked their ObjectTypes and they look identical.
- Yes, the filter is still needed, since there's another `161` types that were not previously fetched and are being added in this filter. Also, it seems to be fetching instances.
- We don't get additional settings instances other than this filter.

@adamro 

This filter added ObjectTypes that weren't fetched before. Upgrading to version V55 causes some of this types to be fetched natively by Salto, and since the filter added the same ObjectType, there was a duplication in certain object types (The ones I mentioned above in my reply to Ori).

adamro commented:
Can we link a relevant repo that was fetched with the previous API and then this API to see the effect?
+ Do we know what we want to do here to reduce noise in the SaaS?

ori-moisis commented on packages/salesforce-adapter/src/filters/settings_type.ts:
```diff
@@ -86,6 +88,7 @@ const filterCreator: FilterCreator = ({ client, config }) => ({
         .map(info => getSettingsTypeName(info.fullName))
         .map(typeName => createSettingsType(client, typeName, knownTypes))
```
IIUC in line 91 we choose to keep the "original" settings type and not the one we create here.
so, why do we need to create it in the first place?
why not `filter` the list of names here before going to salesforce and creating the whole type?

tamtamirr replied:
@ori-moisis 

Tried this approach before, and then noticed the method `createSettingsType` returns more than one ObjectType, so the duplicated types were not filtered out (Assuming they came from another input, I didn't go too deep into the filter)

ori-moisis replied:
use the `knownTypes` map from line 85 then
put the known types in it so they will be used and not re-created

tamtamirr replied:
v, thanks!

tamtamirr commented:
@adamro @ori-moisis 

Just in case, updated the workspace diff with the latest code [here](https://github.com/tamtamirr/my-workspace/pull/2).
P.S made the project private, and added salto-io but not sure you will be able to access the PR yet

tamtamirr commented on packages/salesforce-adapter/src/filters/settings_type.ts:
```diff
@@ -67,6 +66,7 @@ const filterCreator: FilterCreator = ({ client, config }) => ({
    * @param elements
    */
   onFetch: async (elements: Element[]): Promise<FilterResult> => {
+    const objectTypes = elements.filter(isObjectType)
```
will move
